### DISCUSSION TITLE
Git Log Heatmap

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#347] [Announcements]
# Git Log Heatmap

Git Log Heatmap

https://timdeschryver.dev/bits/git-log-heatmap

**Comment by ahmed2m** [comment]
Hey, just a note The copy button doesn't work (at least for me on Firefox) for the code blocks with tabs variant. 
Also, I got the following error on MacOS: `grep: repetition-operator operand invalid`

**Reply by timdeschryver** [reply]
Thanks for bringing this up @ahmed2m , the copy should be fixed.
As for MacOS, can you try the following please (this is a Copilot suggestion):

```bash
git log --since=6.months.ago --pretty=format: --name-only \
    | grep -v '^$' \
    | grep -v '\.(json|lock)$' \
    | sort \
    | uniq -c \
    | sort -rn \
    | head -n 25
```
